src/Adapter/Product/QueryHandler/GetEditableCombinationsListHandler.php

```diff
@@ -118,16 +133,36 @@ private function formatCombinationsForEditing(
                 }
             }
 
-            $combinationsForEditing[] = new CombinationForEditing(
+            $impactOnPrice = new DecimalNumber($combination['price']);
+            $combinationsForEditing[] = new EditableCombinationForListing(
                 $combinationId,
                 $this->buildCombinationName($combinationAttributesInformation),
-                $combinationAttributesInformation
+                $combinationAttributesInformation,
+                (bool) $combination['default_on'],
+                $impactOnPrice,
+                $this->calculateFinalPrice($product, $impactOnPrice),
+                (int) $combination['quantity']
             );
         }
 
         return new CombinationListForEditing($totalCombinationsCount, $combinationsForEditing);
     }
 
+    /**
+     * @param Product $product
+     * @param DecimalNumber $impactOnPrice
+     *
+     * @return DecimalNumber
+     *
+     * @throws NumberExtractorException
+     */
+    private function calculateFinalPrice(Product $product, DecimalNumber $impactOnPrice): DecimalNumber
+    {
+        $productPrice = $this->numberExtractor->extract($product, 'price');
```

Comment:
Product price can be extracted before the loop and be used as a parameter here instead of the product itself
It will avoid creating it at each loop iteration, and since DecimalNumber is immutable it won't be modified